lib/base-chips/field_type.hashed-text.js

```diff
@@ -0,0 +1,44 @@
+var Sealious = require('../main.js');
+var Promise = require("bluebird");
+var crypto = require("crypto");
+
+var field_type_hashed_text = new Sealious.ChipTypes.FieldType({
+	name: "hashed-text",
+	extends: "text",
+	get_description: function(context, params){
+		return "Hash text with chosen algorithm (default md5)";
+	},
+	is_proper_value: function(accept, reject, context, params, new_value){
+		var pattern_array = [];
+		if (params instanceof Object){
+			if (params.digit)
+				pattern_array.push(new RegExp("(.*[0-9]+.*){" + params.capitals + "}"));
+			if (params.capitals)
+				pattern_array.push(new RegExp("(.*[A-Z]+.*){" + params.capitals + "}"));
+
+			var isAccepted = pattern_array.map(n => n.test(new_value)).reduce( (a,b) => a&&b, true);
+
+			isAccepted ? accept() : reject("It didn't fulfill the requirements");
+		}
+		else {
+			accept();
+		}
+	},
+	encode: function(context, params, value_in_code){
+		var salt, algorithm;
+
+		if (params) {
+			salt = params.salt || "" ;
+			algorithm = params.algorithm || "md5";
+		}
+
+		return new Promise(function(resolve, reject){
+			crypto.pbkdf2(value_in_code, salt, 4096, 64, algorithm, function(err, key){
+				if (err) 
+					reject(err)
```

Comment:
I think that conditionals should either be inline and without curly-brackets OR multiline and *with* curly brackets. I know I'm generally not a big believer in code styling nagging, but this construct really rustles my jimmies :)

Why not use the ternary operator you used back in [line 21](https://github.com/Sealious/sealious/pull/245/files#diff-d2c5d00b66474f62251922781664efa5R21)?